### PR TITLE
fix: host app crashes when sending "onError" response for "getUniqueId"

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
- Prevent exception during calling `onError` asynchronously in `getUniqueId`.
- Add an error response for "getUniqueId" to the QA settings section in the demo app.

## Links
- MINI-3385

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/res/layout/qa_settings_activity.xml [hunk 1]
```diff
@@ -80,19 +53,41 @@
...
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:background="@color/bg_section"
+            android:paddingStart="@dimen/medium_16"
+            android:paddingTop="@dimen/small_4"
+            android:paddingEnd="@dimen/medium_16"
+            android:paddingBottom="@dimen/small_4"
+            android:text="@string/lb_unique_id" />
+
+        <com.google.android.material.textfield.TextInputLayout
```
Comment (corycaywood) [inline testapp/src/main/res/layout/qa_settings_activity.xml:77]: I think you should add a toggle switch for enabling the error, similar to how "Access Token" errors work.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt [hunk 2]
```diff
@@ -151,10 +150,16 @@ class MiniAppDisplayActivity : BaseActivity() {
     private fun setupMiniAppMessageBridge() {
         // setup MiniAppMessageBridge
         miniAppMessageBridge = object : MiniAppMessageBridge() {
+
             override fun getUniqueId(
                     onSuccess: (uniqueId: String) -> Unit,
                     onError: (message: String) -> Unit
-            ) = onSuccess(AppSettings.instance.uniqueId)
+            ) {
+                onSuccess(AppSettings.instance.uniqueId)
```
Comment (corycaywood) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:158]: Right now this is always sending success, so it needs to be moved below to the `if/else` which is checking if there's an error.

File: testapp/src/main/res/layout/qa_settings_activity.xml (hunk 1 above)
Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/qa_settings_activity.xml:91]: Yeah, I have added a toggle switch with this input view.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt (hunk 2 above)
Comment (khairul-alam-licon) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt]: Yeah, I have updated this part.

**Review by corycaywood — APPROVED** [review]
LGTM 👍

**Review by munir-rakuten — APPROVED** [review]
looks good to me, thanks 👍

**Pull request merged**